app/controllers/view_components_controller.rb

```diff
@@ -56,7 +56,20 @@ def set_locale
     end
   end
 
+  def template_path
+    if (path = @preview.preview_example_template_path(@example_name))
+      prepend_preview_examples_view_path
+      path
+    else
+      "view_components/preview"
+    end
```

Comment:
Its perhaps a personal style issue but I find assignment in an if condition a bit hard to read

```suggestion
    path = @preview.preview_example_template_path(@example_name)
    if path.presence?
      prepend_preview_examples_view_path
    else
      path = "view_components/preview"
    end
    path
```